src/components/AudioPlayer/AudioPlayer.tsx

```diff
@@ -19,6 +22,7 @@ import Slider from './Slider';
 // import AudioKeyBoardListeners from './AudioKeyboardListeners';
 import MediaSessionApiListeners from './MediaSessionAPIListeners';
 import styles from './AudioPlayer.module.scss';
+import useAudioData from './useAudioData';
```

Comment:
this functionality should not be living here. The components that plays the audio should call the api, extract the url, and then pass than url onto redux.